### PR TITLE
handle statstics screen

### PR DESCRIPTION
**Pull request by khanhVu-ops**
## Related Tickets
- [#Task 78464: [Statistic Screen] create ui, save data local, display data](https://edu-redmine.sun-asterisk.vn/issues/78464)
## WHAT
- 
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift [hunk 1]
```diff
@@ -7,10 +7,124 @@
...
+        state = .watchList
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.register(SearchItemCell.self)
+        tableView.register(EmptyTableViewCell.self)
+    }
+    
+    private func updateUIState() {
+        watchListButton.backgroundColor = state == .watchList ? .color19BAFF : .clear
+        watchListButton.setTitleColor(state == .watchList ? .white : .color19BAFF, for: .normal)
```
Comment (tuta-dev-iotlink) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift]: let isWatchList = state == .watchList

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift [hunk 2]
```diff
@@ -7,10 +7,124 @@
...
+        
+        watchedButton.backgroundColor = state == .watched ? .color19BAFF : .clear
+        watchedButton.setTitleColor(state == .watched ? .white : .color19BAFF, for: .normal)
```
Comment (tuta-dev-iotlink) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift]: tương tự

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift [hunk 3]
```diff
@@ -7,10 +7,124 @@
...
+        numberWatchListLabel.text = "\(numberWatchList)"
+        chartView.colors = [.color19BAFF, .colorFF0B8F]
+        chartView.values = [CGFloat(numberWatched), CGFloat(numberWatchList)]
+        reloadData()
+    }
+    
+    private func reloadData() {
+        mainAsync {
+            self.tableView.reloadData()
+        }
```
Comment (tuta-dev-iotlink) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/View/Statistics/StaticsViewController.swift]: này có lấy weak self chưa nhỉ

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/Apps/AppDelegate.swift [hunk 4]
```diff
@@ -17,5 +18,56 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                try context.save()
+            } catch {
+                // Replace this implementation with code to handle the error appropriately.
+                // fatalError() causes the application to generate a crash log and terminate. You should not use this function in a shipping application, although it may be useful during development.
+                let nserror = error as NSError
+                fatalError("Unresolved error \(nserror), \(nserror.userInfo)")
+            }
+        }
+    }
 }
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/Apps/AppDelegate.swift]: xoá comment đi e

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+import Foundation
+import UIKit
+import CoreData
+
+final class CoreDataStatistics {
+    static let shared = CoreDataStatistics()
+    
+    func save(item: SearchModel, isWatched: Bool) {
+        let model = NSEntityDescription.insertNewObject(forEntityName: "MovieLocalModel", into: CoreDataUtils.shared.privateManagedObjectContext) as! MovieLocalModel
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift]: force unwrap, convention

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    
+    func fetch(isWatched: Bool) -> [MovieLocalModel] {
+       
+        let fetchRequest: NSFetchRequest<MovieLocalModel> = MovieLocalModel.fetchRequest()
+        let p1 = NSPredicate(format: "isWatched == %@", NSNumber(value: isWatched))
+
+        fetchRequest.predicate = NSCompoundPredicate.init(type: .and, subpredicates: [
+            p1
+        ])
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift]: không đặt tên p1 , sửa lại e

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    
+    func delete(item: SearchModel, isWatched: Bool) {
+        let fetchRequest: NSFetchRequest<MovieLocalModel> = MovieLocalModel.fetchRequest()
+        let p1 = NSPredicate(format: "isWatched == %@", NSNumber(value: isWatched))
+        let p2 = NSPredicate(format: "id == %@ ", NSNumber(value: item.id))
+        
+        fetchRequest.predicate = NSCompoundPredicate.init(type: .and, subpredicates: [
+            p1,
+            p2,
+        ])
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift]: tương tự

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    
+    func delete(item: SearchModel, isWatched: Bool) {
+        let fetchRequest: NSFetchRequest<MovieLocalModel> = MovieLocalModel.fetchRequest()
+        let query1 = NSPredicate(format: "isWatched == %@", NSNumber(value: isWatched))
+        let query2 = NSPredicate(format: "id == %@ ", NSNumber(value: item.id))
+        
+        fetchRequest.predicate = NSCompoundPredicate.init(type: .and, subpredicates: [
+            query1,
+            query2,
+        ])
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/CoreData/CoreDataStatistics.swift]: đặt lại tên biến đi e

File: HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/Utils/CommonView/PieChartView.swift [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+            let endAngle = startAngle + (2 * .pi * (value / total))
+            
+            // Set the fill color
+            context.setFillColor(colors[index].cgColor)
+            
+            // Move to the center
+            context.move(to: center)
+            
+            // Add the arc
+            context.addArc(center: center, radius: radius, startAngle: startAngle, endAngle: endAngle, clockwise: false)
```
Comment (ghost) [inline HN_Pra19_IOS_Movie_App/HN_Pra19_IOS_Movie_App/Utils/CommonView/PieChartView.swift]: convention

**Review by ghost — APPROVED** [review]

**Pull request merged**